### PR TITLE
[issue #982 & issue #978]enhance nacos config

### PR DESCRIPTION
**Pull request by zkzlx**
### Describe what this PR does / why we need it
Solve three problems:
1. nacos config shared-dataids cont select group
2. shared-dataids and ext-config can support profile recognition
3. fix part of original unreasonable design

### Does this pull request fix one issue?

#982
#978

### CODE REVIEW

File: spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/client/NacosPropertySource.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
 
 	NacosPropertySource(String group, String dataId, Map<String, Object> source,
 			Date timestamp, boolean isRefreshable) {
-		super(dataId, source);
+		super(String.format("%s#%s", dataId, group), source);
```
Comment (fangjian0423) [inline spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/client/NacosPropertySource.java]: "#" could be extractd in `NacosConfigProperties`

File: spring-cloud-alibaba-nacos-config/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 2]
```diff
@@ -3,7 +3,6 @@
     {
       "name": "spring.cloud.nacos.server-addr",
       "type": "java.lang.String",
-      "defaultValue": "localhost:8848",
```
Comment (fangjian0423) [inline spring-cloud-alibaba-nacos-config/src/main/resources/META-INF/additional-spring-configuration-metadata.json]: default server-addr is localhost:8848, why remove it?

File: spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/refresh/NacosRefreshHistory.java [hunk 3]
```diff
@@ -16,26 +16,55 @@
...
-		@Override
-		protected DateFormat initialValue() {
-			return new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+	private MessageDigest md;
+
+	public NacosRefreshHistory() {
+		try {
+			md = MessageDigest.getInstance("MD5");
+		}
+		catch (NoSuchAlgorithmException ignored) {
```
Comment (fangjian0423) [inline spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/refresh/NacosRefreshHistory.java]: add error log

File: spring-cloud-alibaba-nacos-config/src/main/resources/META-INF/additional-spring-configuration-metadata.json (hunk 2 above)
Comment (zkzlx) [reply spring-cloud-alibaba-nacos-config/src/main/resources/META-INF/additional-spring-configuration-metadata.json]: code was accidentally lost while merging, I'll recover it

File: spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/NacosConfigAutoConfiguration.java [hunk 4]
```diff
@@ -54,13 +54,24 @@ public NacosRefreshHistory nacosRefreshHistory() {
...
-			NacosConfigProperties configProperties,
+			NacosConfigManager nacosConfigManager,
 			NacosRefreshProperties nacosRefreshProperties,
-			NacosRefreshHistory refreshHistory) {
-		return new NacosContextRefresher(nacosRefreshProperties, refreshHistory,
-				configProperties.configServiceInstance());
+			NacosRefreshHistory nacosRefreshHistory) {
+		// Compatible with older configurations
+		if (nacosConfigManager.getNacosConfigProperties().isRefreshEnabled()
+				&& null != nacosRefreshProperties
```
[the remark this answers is not in the document]
Comment (zkzlx) [reply spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/NacosConfigAutoConfiguration.java]: Compatible with older configurations, may be previous configurations

[the remark this answers is not in the document]
Comment (zkzlx) [reply spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/NacosConfigAutoConfiguration.java]: Consider that it is not necessary to be compatible with the previous configuration, and use the new configuration if necessary

File: spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/NacosConfigAutoConfiguration.java [hunk 5]
```diff
@@ -54,13 +54,19 @@ public NacosRefreshHistory nacosRefreshHistory() {
...
 	@Bean
 	public NacosContextRefresher nacosContextRefresher(
-			NacosConfigProperties configProperties,
-			NacosRefreshProperties nacosRefreshProperties,
-			NacosRefreshHistory refreshHistory) {
-		return new NacosContextRefresher(nacosRefreshProperties, refreshHistory,
-				configProperties.configServiceInstance());
+			NacosConfigManager nacosConfigManager,
+			NacosRefreshHistory nacosRefreshHistory) {
+		// Consider that it is not necessary to be compatible with the previous configuration
```
Comment (fangjian0423) [inline spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/NacosConfigAutoConfiguration.java]: It seems the comments should be removed?

**Pull request merged**